script/SetCancellers.s.sol

```diff
@@ -0,0 +1,108 @@
+
+        // Get the timelock address from the governor
+        address timelockAddress = governor.timelock();
+        console.log("Timelock address:", timelockAddress);
+
+        // Get the timelock contract
+        TimelockControllerUpgradeable timelock = TimelockControllerUpgradeable(payable(timelockAddress));
+
+        // Get the access control contract
+        AccessControlEnumerableUpgradeable accessControl = AccessControlEnumerableUpgradeable(payable(timelockAddress));
+
+        // Make sure caller has admin role as we're about to modify the timelock
+        address caller = vm.addr(deployerPrivateKey);
+        require(timelock.hasRole(TIMELOCK_ADMIN_ROLE, caller), "Caller does not have admin role on timelock");
+
+        // Step 1: Grant CANCELLER_ROLE to the three L2 Safes (if they don't already have it)
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_1)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 1:", L2_SAFE_1);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_1);
+        } else {
+            console.log("L2 Safe 1 already has CANCELLER_ROLE:", L2_SAFE_1);
+        }
+
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_2)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 2:", L2_SAFE_2);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_2);
+        } else {
+            console.log("L2 Safe 2 already has CANCELLER_ROLE:", L2_SAFE_2);
+        }
+
+        if (!timelock.hasRole(CANCELLER_ROLE, L2_SAFE_3)) {
+            console.log("Granting CANCELLER_ROLE to L2 Safe 3:", L2_SAFE_3);
+            timelock.grantRole(CANCELLER_ROLE, L2_SAFE_3);
+        } else {
+            console.log("L2 Safe 3 already has CANCELLER_ROLE:", L2_SAFE_3);
+        }
+
+        // Step 2: Ensure all proposers have CANCELLER_ROLE (which is the default in OpenZeppelin's TimelockController)
+        uint256 proposerCount = accessControl.getRoleMemberCount(PROPOSER_ROLE);
+        console.log("Found", proposerCount, "proposers");
+
+        for (uint256 i = 0; i < proposerCount; i++) {
+            address proposer = accessControl.getRoleMember(PROPOSER_ROLE, i);
+            if (!timelock.hasRole(CANCELLER_ROLE, proposer)) {
+                console.log("Granting CANCELLER_ROLE to proposer:", proposer);
+                timelock.grantRole(CANCELLER_ROLE, proposer);
+            } else {
+                console.log("Proposer already has CANCELLER_ROLE:", proposer);
+            }
+        }
+
+        // Verify the results
+        console.log("Verification:");
+        console.log("L2 Safe 1 has CANCELLER_ROLE:", timelock.hasRole(CANCELLER_ROLE, L2_SAFE_1));
+        console.log("L2 Safe 2 has CANCELLER_ROLE:", timelock.hasRole(CANCELLER_ROLE, L2_SAFE_2));
+        console.log("L2 Safe 3 has CANCELLER_ROLE:", timelock.hasRole(CANCELLER_ROLE, L2_SAFE_3));
+
+        // Verify all proposers have the CANCELLER_ROLE
```

Comment:
Same here why we check for all addresses? Wouldnt it make sense to check if the 3 safes also have/dont have the "Proposer" role?